sql/planbuilder/ddl.go

```diff
@@ -1065,7 +1065,9 @@ func (b *Builder) buildAlterAutoIncrement(inScope *scope, ddl *ast.DDL, table *p
 func (b *Builder) buildAlterNotNull(inScope *scope, ddl *ast.DDL, table *plan.ResolvedTable) (outScope *scope) {
 	outScope = inScope
 	spec := ddl.NotNullSpec
-	for _, c := range table.Schema() {
+
+	resolvedSchema := b.resolveSchemaDefaults(inScope, table.Schema())
```

Comment:
Might make a note for these why they're necessary, since the tests that require them aren't in this package.